### PR TITLE
feat: Added the experimental-require-slot-types rule

### LINKED ISSUE #347: New rule: `require-slot-types`
### Motivation

It is possible to add types to component slots and their props:

```
<script lang="ts">
    interface $$Slots {
        default: { prop: boolean; };
    }
</script>

<slot prop={true} />
```

I would like to propose a rule to enforce these slot types.

### Description

The rule should report any component that includes a `<slot>` and does not declare the `$$Slots` interface.

### Examples

```svelte
<!-- ✓ GOOD -->
<script>
</script>

<b>No slots here!</b>



<script>
  interface $$Slots {
    default: Record<string, never>;
  }
</script>

<slot />




<script lang="ts">
    interface $$Slots {
        default: { prop: boolean; };
    }
</script>

<slot prop={true} />


<script lang="ts">
    interface $$Slots {
        named: Record<string, never>;
    }
</script>

<slot name = "named" />

<!-- ✗ BAD -->
<script>
</script>

<slot />
```

### Additional comments

The relevant [RFC](https://github.com/dummdidumm/rfcs/blob/ts-typedefs-within-svelte-components/text/ts-typing-props-slots-events.md#typing-slots).

### PR DESCRIPTION
**Pull request by marekdedic**
Closes #347

Depends on #365 being merged first - it introduces the experimental rules section

### CODE REVIEW

**Comment by marekdedic** [comment]
Hey, @ota-meshi, there's something broken with prettier and the line

https://github.com/ota-meshi/eslint-plugin-svelte/blob/f38495d1ae07a261dfc3ae7f9803bb0abf021399/tools/render-rules.ts#L14

It keeps switching back and forth whether it wants parentheses or not, event when there are no changes to the file in question

**Comment by ota-meshi** [comment]
I think `Prettier` changed the format way in a minor version. I think you and we need to remove `yarn.lock` and `node_modules` and re-install.

**Comment by marekdedic** [comment]
Ok, ok, I think this PR will pass now and I will just pull master & remove node_modules then. Thanks

**Comment by marekdedic** [comment]
... or not :D

File: docs/rules/experimental-require-slot-types.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+The  `$$Slots` interface is experimental and is documented in [svelte RFC #38](https://github.com/dummdidumm/rfcs/blob/ts-typedefs-within-svelte-components/text/ts-typing-props-slots-events.md#typing-slots).
+
+<ESLintCodeBlock>
+
+<!--eslint-skip-->
+
+```svelte
+<!-- ✓ GOOD -->
+<script>
+  // eslint svelte/experimental-require-slot-types: "error"
```
Comment (ota-meshi) [inline docs/rules/experimental-require-slot-types.md:26]: Could you change it to use block comments?

```suggestion
  /* eslint svelte/experimental-require-slot-types: "error" */
```

https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments

File: docs/rules/experimental-require-slot-types.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+</ESLintCodeBlock>
+
+<ESLintCodeBlock>
+
+<!--eslint-skip-->
+
+```svelte
+<!-- ✓ GOOD -->
+<script>
+  // eslint svelte/experimental-require-slot-types: "error"
```
Comment (ota-meshi) [inline docs/rules/experimental-require-slot-types.md:41]:
```suggestion
  /* eslint svelte/experimental-require-slot-types: "error" */
```

File: docs/rules/experimental-require-slot-types.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+</ESLintCodeBlock>
+
+<ESLintCodeBlock>
+
+<!--eslint-skip-->
+
+```svelte
+<!-- ✓ GOOD -->
+<script lang="ts">
+  // eslint svelte/experimental-require-slot-types: "error"
```
Comment (ota-meshi) [inline docs/rules/experimental-require-slot-types.md:60]:
```suggestion
  /* eslint svelte/experimental-require-slot-types: "error" */
```

File: docs/rules/experimental-require-slot-types.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
```
Comment (ota-meshi) [inline docs/rules/experimental-require-slot-types.md:79]:
```suggestion
  /* eslint svelte/experimental-require-slot-types: "error" */
```

File: docs/rules/experimental-require-slot-types.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+</ESLintCodeBlock>
+
+<ESLintCodeBlock>
+
+<!--eslint-skip-->
+
+```svelte
+<!-- ✗ BAD -->
+<script>
+  // eslint svelte/experimental-require-slot-types: "error"
```
Comment (ota-meshi) [inline docs/rules/experimental-require-slot-types.md:98]:
```suggestion
  /* eslint svelte/experimental-require-slot-types: "error" */
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: docs/rules/experimental-require-slot-types.md (hunk 1 above)
Comment (marekdedic) [reply docs/rules/experimental-require-slot-types.md:26]: Of course :) But I think it should work either way ;)

Comment (ota-meshi) [reply docs/rules/experimental-require-slot-types.md]: I think it only works with block comments.  Thanks for the change anyway.

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**